libs/community/langchain_community/embeddings/textembed.py

```diff
@@ -0,0 +1,390 @@
+            Embeds a list of documents.
+        aembed_documents(self, texts: List[str]) -> List[List[float]]:
+            Asynchronously embeds a list of documents.
+        embed_query(self, text: str) -> List[float]:
+            Embeds a single query.
+        aembed_query(self, text: str) -> List[float]:
+            Asynchronously embeds a single query.
+
+    Example:
+        .. code-block:: python
+
+            from langchain_community.embeddings import TextEmbedEmbeddings
+            embeddings = TextEmbedEmbeddings(
+                model="sentence-transformers/clip-ViT-B-32",
+                textembed_api_url="http://localhost:8000/v1",
+                textembed_api_key="<API_KEY>"
+            )
+    """
+
+    model: str
+    "Underlying TextEmbed model id."
+
+    textembed_api_url: str = "http://localhost:7997"
+    """Endpoint URL to use."""
+
+    textembed_api_key: str = "None"
+    """API Key for authentication"""
+
+    client: Any = None
+    """TextEmbed client."""
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+
+    @root_validator(allow_reuse=True)
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that api key and URL exist in the environment.
+
+        Args:
+            values (Dict): Dictionary of values to validate.
+
+        Returns:
+            Dict: Validated values.
+        """
+        values["textembed_api_url"] = get_from_dict_or_env(
+            values, "textembed_api_url", "TEXTEMBED_API_URL"
+        )
+
+        values["textembed_api_key"] = get_from_dict_or_env(
+            values, "textembed_api_key", "TEXTEMBED_API_KEY"
+        )
+
+        values["client"] = AsyncOpenAITextEmbedEmbeddingClient(
+            host=values["textembed_api_url"], api_key=values["textembed_api_key"]
+        )
+        return values
+
+    def embed_documents(self, texts: List[str]) -> List[List[float]]:
+        """Call out to TextEmbed's embedding endpoint.
+
+        Args:
+            texts (List[str]): The list of texts to embed.
+
+        Returns:
+            List[List[float]]: List of embeddings, one for each text.
+        """
+        embeddings = self.client.embed(
+            model=self.model,
+            texts=texts,
+        )
+        return embeddings
+
+    async def aembed_documents(self, texts: List[str]) -> List[List[float]]:
+        """Async call out to TextEmbed's embedding endpoint.
+
+        Args:
+            texts (List[str]): The list of texts to embed.
+
+        Returns:
+            List[List[float]]: List of embeddings, one for each text.
+        """
+        embeddings = await self.client.aembed(
+            model=self.model,
+            texts=texts,
+        )
+        return embeddings
+
+    def embed_query(self, text: str) -> List[float]:
+        """Call out to TextEmbed's embedding endpoint for a single query.
+
+        Args:
+            text (str): The text to embed.
```

Comment:
we usually don't document the type in the param strings since we use type annotations

Reply:
Hey @eyurtsev 

Thank you for your feedback. I understand that we typically don't include the type in the parameter strings since we use type annotations. However, I believe including the type in the docstring can be beneficial for a few reasons:

Readability: When someone is reading the docstring, they don't need to look at the function signature to understand the expected types.
Documentation Tools: Some documentation tools generate more detailed and user-friendly documentation when the types are included in the docstring.
Consistency: It can provide a consistent format for those who might not be as familiar with type annotations.